fsync.c

```diff
@@ -0,0 +1,104 @@
+#include "redisraft.h"
+#include <pthread.h>
+#include <string.h>
+#include <errno.h>
+
+void fsyncThreadAddTask(FsyncThread *th, int fd, raft_index_t requested_index)
+{
+    int rc;
+
+    pthread_mutex_lock(&th->mtx);
+
+    th->requested_index = requested_index;
+    th->fd = fd;
+    th->need_fsync = 1;
+    th->running = 1;
+
+    rc = pthread_cond_signal(&th->cond);
+    RedisModule_Assert(rc == 0);
+
+    pthread_mutex_unlock(&th->mtx);
+}
+
+/* wait fsync thread until it completes fsync() call */
+void fsyncThreadWaitUntilCompleted(FsyncThread *th)
+{
+    pthread_mutex_lock(&th->mtx);
+    while (th->running == 1) {
+        pthread_cond_wait(&th->cond, &th->mtx);
+    }
+    pthread_mutex_unlock(&th->mtx);
+}
+
+static void* fsyncLoop(void *arg)
+{
+    int rc, fd;
+    raft_index_t request_idx;
+    FsyncThread *th = arg;
+
+    while (1) {
+        pthread_mutex_lock(&th->mtx);
+        while (th->need_fsync == 0) {
+            rc = pthread_cond_wait(&th->cond, &th->mtx);
```

Comment:
EINTR probably needs to be gracefully ignored here.

Reply:
`pthread_cond_wait()` will not return EINTR according to POSIX (https://pubs.opengroup.org/onlinepubs/7908799/xsh/pthread_cond_wait.html) 


> These functions will not return an error code of [EINTR].

In our case, pthread_cond_wait() never returns an error (if it does, we have a bug).

Reply:
Wasn't aware of that!